### ISSUE TITLE
Paste HTML Code into NGEN Editor shows unexpected results after save

### ISSUE THREAD

**Issue by thilgertbo** [getgrav/grav-premium-issues#7]
If I paste HTML code into the NGEN Editor some parts get short codes formatting after pushing save.
The system try to guess what I pasted and change the HTML code into code blocks.
```html
<div class="py-12 bg-white">

  <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">

<div class="lg:text-center">

  <h2 class="text-base text-indigo-600 font-semibold tracking-wide uppercase">Unsere Dienstleistungen</h2>

  <p class="mt-2 text-3xl leading-8 font-extrabold tracking-tight text-gray-900 sm:text-4xl">

der beste Weg für eine stabile IT 

  </p>

  <p class="mt-4 max-w-2xl text-xl text-gray-500 lg:mx-auto">

Der Betrieb von IT Hardware und Software wird von uns effizient, stabil und kostengünstig durchgeführt. 

  </p>

</div>

<div class="mt-10">

  <dl class="space-y-10 md:space-y-0 md:grid md:grid-cols-2 md:gap-x-8 md:gap-y-10">

<div class="flex">

  <div class="flex-shrink-0">

<div class="flex items-center justify-center h-12 w-12 rounded-md bg-indigo-500 text-white">



  <!-- Heroicon name: globe-alt -->



  <svg class="h-6 w-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">



    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M21 12a9 9 0 01-9 9m9-9a9 9 0 00-9-9m9 9H3m9 9a9 9 0 01-9-9m9 9c1.657 0 3-4.03 3-9s-1.343-9-3-9m0 18c-1.657 0-3-4.03-3-9s1.343-9 3-9m-9 9a9 9 0 019-9" />



  </svg>



</div>

  </div>

  <div class="ml-4">

<dt class="text-lg leading-6 font-medium text-gray-900">



  Competitive exchange rates



</dt>



<dd class="mt-2 text-base text-gray-500">



  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.



</dd>

  </div>

</div>

<div class="flex">

  <div class="flex-shrink-0">

<div class="flex items-center justify-center h-12 w-12 rounded-md bg-indigo-500 text-white">



  <!-- Heroicon name: scale -->



  <svg class="h-6 w-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">



    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M3 6l3 1m0 0l-3 9a5.002 5.002 0 006.001 0M6 7l3 9M6 7l6-2m6 2l3-1m-3 1l-3 9a5.002 5.002 0 006.001 0M18 7l3 9m-3-9l-6-2m0-2v2m0 16V5m0 16H9m3 0h3" />



  </svg>



</div>

  </div>

  <div class="ml-4">

<dt class="text-lg leading-6 font-medium text-gray-900">



  No hidden fees



</dt>



<dd class="mt-2 text-base text-gray-500">



  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.



</dd>

  </div>

</div>

<div class="flex">

  <div class="flex-shrink-0">

<div class="flex items-center justify-center h-12 w-12 rounded-md bg-indigo-500 text-white">



  <!-- Heroicon name: lightning-bolt -->



  <svg class="h-6 w-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">



    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M13 10V3L4 14h7v7l9-11h-7z" />



  </svg>



</div>

  </div>

  <div class="ml-4">

<dt class="text-lg leading-6 font-medium text-gray-900">



  Transfers are instant



</dt>



<dd class="mt-2 text-base text-gray-500">



  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.



</dd>

  </div>

</div>

<div class="flex">

  <div class="flex-shrink-0">

<div class="flex items-center justify-center h-12 w-12 rounded-md bg-indigo-500 text-white">



  <!-- Heroicon name: annotation -->



  <svg class="h-6 w-6" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" aria-hidden="true">



    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M7 8h10M7 12h4m1 8l-4-4H5a2 2 0 01-2-2V6a2 2 0 012-2h14a2 2 0 012 2v8a2 2 0 01-2 2h-3l-4 4z" />



  </svg>



</div>

  </div>

  <div class="ml-4">

<dt class="text-lg leading-6 font-medium text-gray-900">

  Mobile notifications

</dt>

<dd class="mt-2 text-base text-gray-500">

  Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.

</dd>

  </div>

</div>

  </dl>

</div>

  </div>

</div>
```

**Comment by rhukster** [comment]
Can you provide an example of the HTML code you pasted, and how it looks after save?

**Comment by rhukster** [comment]
Perhaps you could paste the HTML into a code block? https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/creating-and-highlighting-code-blocks#fenced-code-blocks 

That way we can test ourselves. Thanks!

**Comment by rhukster** [comment]
That is your content? NextGen handles pasting in 'rich' text pretty well.  For example, the browser view of this:

![HTML Example  Grav 2020-12-19 at 6 53 26 PM](https://user-images.githubusercontent.com/1084697/102703404-8528c280-422b-11eb-9263-d4422bf98dc7.jpg)

Then selecting and copying:

![HTML Example  Grav 2020-12-19 at 6 54 12 PM](https://user-images.githubusercontent.com/1084697/102703410-a2f62780-422b-11eb-8f25-fc548bc746fa.jpg)

Then pasting into Nextgen:

![HTML Example ( pluginsckeditorhtml-paste-7 )  Grav 2020-12-19 at 6 56 05 PM](https://user-images.githubusercontent.com/1084697/102703434-df298800-422b-11eb-892c-e4ff14f9d5e0.jpg)

Then on "Save" you get:

![HTML Example ( pluginsckeditorhtml-paste-7 )  Grav 2020-12-19 at 6 56 54 PM](https://user-images.githubusercontent.com/1084697/102703444-02543780-422c-11eb-8423-1c01ed42a72b.jpg)

This is expected as NexGen is converting the HTML into Markdown, which has simpler markup  than HTML does. 

You can recreate multiple columns and blocks with shortcodes, but frankly that's not really what content is intended to do. 

That example you pasted, and I recreated by hand, is better handled by a custom modular layout, which is what is already happening in Typhoon theme with the 'features' modular type.  This way you have a list of items that have dedicated fields.

![Modular Page  Grav 2020-12-19 at 7 01 57 PM](https://user-images.githubusercontent.com/1084697/102703492-b6ee5900-422c-11eb-90fb-eb1d76511acc.jpg)

And the admin UI for it:

![Homepage Highlights ( homemodular_highlights )  Grav 2020-12-19 at 7 03 00 PM](https://user-images.githubusercontent.com/1084697/102703505-d84f4500-422c-11eb-9df6-fff9b32aabbb.jpg)

**Comment by thilgertbo** [comment]
I got that, I just play around with the tailwind css library. And for learning and understanding I just copy an example of the code base there.  Sometimes I need the possibility to add plain HTML into a content. It makes life a lot easier to copy and paste code blocks. May you should think about a shortcode in inject plain html into the content.

**Comment by rhukster** [comment]
That a great idea 👍🏻

**Comment by w00fz** [comment]
hey @thilgertbo , just wanted to let you know that the new version of NextGen that has just been released (v1.0.3), now includes a new HTML Snippets feature that will make having html code in content a lot more manageable.

Have a look at the updated documentation: https://getgrav.org/premium/nextgen-editor/docs#html-snippets
